### PR TITLE
docs: add header component

### LINKED ISSUE #924: components.header undocumented
#### Do you want to request a _feature_ or report a _bug_?
_question_

Currently TimeGridHeader makes use of a header component that can be passed from Calendar. https://github.com/intljusticemission/react-big-calendar/blob/d2e02c4bfde41b39ea22db344cbd8b948b17fc8b/src/TimeGridHeader.js#L70

The property is however [not documented](http://intljusticemission.github.io/react-big-calendar/examples/index.html#prop-components). Is this indeed only an implementation detail or meant for actual usage? Currently this seems to be the only way to render a custom header for days in the week view.

### PR DESCRIPTION
**Pull request by eps1lon**
Closes #924 

I added some additional information about where this component is used. Since I'm not familiar with this package this might be incomplete.

### CODE REVIEW

File: src/Calendar.js [hunk 1]
```diff
@@ -649,6 +649,10 @@ class Calendar extends React.Component {
       eventContainerWrapper: elementType,
       dayWrapper: elementType,
       dateCellWrapper: elementType,
+      /**
+       * component used as a header for each column in the TimeGridHeader
+       */
+      header: elementType,
```
Comment (jquense) [inline src/Calendar.js]: actually this is already documented below in the specific views segment of this proptype

Comment (eps1lon) [reply src/Calendar.js]: I'm not sure I follow. What's holding this PR back?

Comment (jquense) [reply src/Calendar.js]: The PR isn't needed the header component is already documented here

**Comment by jquense** [comment]
Thanks for the PR, the docs have this covered already it looks like! https://intljusticemission.github.io/react-big-calendar/examples/index.html#prop-components

**Comment by eps1lon** [comment]
It is currently possible however to pass `components={{ header: () => 'GlobalHeader' }}`

**Pull request closed without merging**